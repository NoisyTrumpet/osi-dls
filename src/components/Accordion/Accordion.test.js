import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import { Accordion } from "./";

describe("<Accordion />", () => {
  test("should render", () => {
    const wrapper = shallow(<Accordion />);
    expect(wrapper.getElement()).toBeTruthy();
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  test("should render accordion with 3 sections", () => {
    const Acc = (
      <Accordion>
        <Accordion.Section
          title="Section 1"
          titleID="accordion1id"
          panelId="sect1"
        >
          <span>Text 1</span>
          <span>Text 2</span>
        </Accordion.Section>
        <Accordion.Section
          title="Section 2"
          titleID="accordion2id"
          panelId="sect2"
        >
          <img
            src="https://webfileswest.uswest.os.ctl.io/corporate.hallmark.com/wp-content/uploads/2018/06/Hallmark-Greetings-Logo-black-160x160.jpg"
            alt="Hallmark"
          />
        </Accordion.Section>
        <Accordion.Section
          title="Section Link"
          titleID="accordion4id"
          panelId="sect4"
          link="https://osilife.com/"
        />
      </Accordion>
    );
    const wrapper = shallow(Acc);
    expect(wrapper.getElement()).toBeTruthy();
    expect(wrapper.find(Accordion.Section).length).toBe(3);
    expect(wrapper.find("span").length).toBe(2);
    expect(wrapper.find("img").length).toBe(1);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
