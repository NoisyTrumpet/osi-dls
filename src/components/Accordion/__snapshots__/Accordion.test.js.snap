// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Accordion /> should render 1`] = `
<div
  className="accordion"
  onClick={[Function]}
  role="presentation"
>
  <ContextProvider
    value={null}
  />
</div>
`;

exports[`<Accordion /> should render accordion with 3 sections 1`] = `
<div
  className="accordion"
  onClick={[Function]}
  role="presentation"
>
  <ContextProvider
    value={null}
  >
    <AccordionSection
      panelId="sect1"
      title="Section 1"
      titleID="accordion1id"
      titleVariant="headline-6"
    >
      <span>
        Text 1
      </span>
      <span>
        Text 2
      </span>
    </AccordionSection>
    <AccordionSection
      panelId="sect2"
      title="Section 2"
      titleID="accordion2id"
      titleVariant="headline-6"
    >
      <img
        alt="Hallmark"
        src="https://webfileswest.uswest.os.ctl.io/corporate.hallmark.com/wp-content/uploads/2018/06/Hallmark-Greetings-Logo-black-160x160.jpg"
      />
    </AccordionSection>
    <AccordionSection
      link="https://www.hallmark.com/"
      panelId="sect4"
      title="Section Link"
      titleID="accordion4id"
      titleVariant="headline-6"
    />
  </ContextProvider>
</div>
`;
